pkg/controller/discovery/container/pod.go

```diff
@@ -0,0 +1,171 @@
+package container
+
+import (
+	"context"
+	"github.com/fusor/mig-controller/pkg/controller/discovery/model"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+//
+// A collection of k8s Pod resources.
+type PodCollection struct {
+	// Base
+	BaseCollection
+}
+
+func (r *PodCollection) AddWatch(dsController controller.Controller) error {
+	err := dsController.Watch(
+		&source.Kind{
+			Type: &v1.Pod{},
+		},
+		&handler.EnqueueRequestForObject{},
+		r)
+	if err != nil {
+		Log.Trace(err)
+		return err
+	}
+
+	return nil
+}
+
+func (r *PodCollection) Reconcile() error {
+	mark := time.Now()
+	sr := SimpleReconciler{Db: r.ds.Container.Db}
+	err := sr.Reconcile(r)
+	if err != nil {
+		Log.Trace(err)
+		return err
+	}
+	r.hasReconciled = true
+	Log.Info(
+		"PodCollection reconciled.",
+		"cluster",
+		r.ds.cluster,
+		"duration",
+		time.Since(mark))
+
+	return nil
+}
+
+func (r *PodCollection) GetDiscovered() ([]model.Model, error) {
+	models := []model.Model{}
+	onCluster := v1.PodList{}
+	err := r.ds.client.List(context.TODO(), nil, &onCluster)
+	if err != nil {
+		Log.Trace(err)
+		return nil, err
+	}
+	for _, discovered := range onCluster.Items {
+		ns := &model.Pod{
+			Base: model.Base{
+				Cluster: r.ds.cluster.PK,
+			},
+		}
+		ns.With(&discovered)
+		models = append(models, ns)
+	}
+
+	return models, nil
+}
+
+func (r *PodCollection) GetStored() ([]model.Model, error) {
+	models := []model.Model{}
+	list, err := r.ds.cluster.PodList(r.ds.Container.Db, nil)
+	if err != nil {
+		Log.Trace(err)
+		return nil, err
+	}
+	for _, pod := range list {
+		models = append(models, pod)
+	}
+
+	return models, nil
+}
+
+//
+// Predicate methods.
+//
+
+func (r *PodCollection) Create(e event.CreateEvent) bool {
+	Log.Reset()
+	object, cast := e.Object.(*v1.Pod)
+	if !cast {
+		return false
+	}
+	pod := model.Pod{
+		Base: model.Base{
+			Cluster: r.ds.cluster.PK,
+		},
+	}
+	pod.With(object)
+	pod.SetPk()
+	tx, _ := r.ds.Container.Db.Begin()
+	err := pod.Insert(tx)
+	if err != nil {
+		tx.Rollback()
+		Log.Trace(err)
+	} else {
+		tx.Commit()
+	}
+
+	return false
+}
+
+func (r *PodCollection) Update(e event.UpdateEvent) bool {
+	Log.Reset()
+	object, cast := e.ObjectNew.(*v1.Pod)
+	if !cast {
+		return false
+	}
+	pod := model.Pod{
+		Base: model.Base{
+			Cluster: r.ds.cluster.PK,
+		},
+	}
+	pod.With(object)
+	pod.SetPk()
+	tx, _ := r.ds.Container.Db.Begin()
+	err := pod.Update(tx)
+	if err != nil {
+		tx.Rollback()
+		Log.Trace(err)
+	} else {
+		tx.Commit()
+	}
+
+	return false
+}
+
+func (r *PodCollection) Delete(e event.DeleteEvent) bool {
+	Log.Reset()
+	object, cast := e.Object.(*v1.Pod)
+	if !cast {
+		return false
+	}
+	pod := model.Pod{
+		Base: model.Base{
+			Cluster: r.ds.cluster.PK,
+		},
+	}
+	pod.With(object)
+	pod.SetPk()
+	tx, _ := r.ds.Container.Db.Begin()
```

Comment:
Is it safe to skip the error check here?

Reply:
Fixed.
Updated model methods to accept the DB as an _interface_ and removed all but (1) explicit use of transaction.  Handling the error in the remaining place.  Also added checking for tx.Commit() errors.